lib/ansible/modules/cloud/amazon/aws_ssm_send_command.py

```diff
@@ -0,0 +1,230 @@
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: aws_ssm_send_command
+short_description: Execute commands through Simple System Manager (SSM) a.k.a. Run Command
+description:
+  - This module allows you to execute commands through SSM/Run Command.
+version_added: "2.9"
+author: "Joe Wozniak (@woznij)"
+requirements:
+  - python >= 2.6
+  - boto3
+notes:
+  - Async invocation will always return an empty C(output) key.
+  - Synchronous invocation may result in a function timeout, resulting in an
+    empty C(output) key.
+options:
+  name:
+    description:
+      - This should match the name of the SSM document to be invoked.
+    type: str
+    required: true
+  comment:
+    description:
+      - A comment about this particular invocation.
+    required: false
+    type: str
+  instance_ids:
+    description:
+      - A list of instance IDs for the instances you wish to run this command
+        document against.
+    required: true
+    type: list
+  wait:
+    description:
+      - Whether to wait for the function results or not. If I(wait) is false,
+        the task will not return any results. To wait for the command to
+        complete, set C(wait=true) and the result will be available in the
+        I(output) key.
+    required: false
+    type: bool
+    default: true
+  parameters:
+    description:
+      - A dictionary to be provided as the parameters for the SSM document
+        you're invoking.
+    required: false
+    default: {}
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+- aws_ssm_send_command:
+    name: AWS-UpdateSSMAgent
+    comment: "SSM agent update check"
+    instance_ids:
+      - i-123987193812
+      - i-289189288278
+    parameters:
+      version: latest
+      allowDowngrade: 'false'
+    wait: true
+  register: response
+'''
+
+RETURN = '''
+output:
+    description: If wait=true, will return the output of the executed command. Sample truncated for brevity.
+    returned: success
+    type: str
+    sample: "Updating amazon-ssm-agent from 2.3.539.0 to latest\nSuccessfully downloaded https://s3.us-west-2.amazonaws.com/amazon-ssm-us-west-2/ssm-agent..."
+status:
+    description: Status of the run command.
+    returned: success
+    type: str
+    sample: Success
+'''
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import boto3_conn, ec2_argument_spec, get_aws_connection_info, AWSRetry
+import traceback
+from time import sleep
+
+try:
+    import boto3
+    import botocore
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+@AWSRetry.backoff(tries=5, delay=5, backoff=2.0)
+def ssm_send_command(conn, **kwargs):
+    return conn.send_command(**kwargs)
+
+
+@AWSRetry.backoff(tries=5, delay=5, backoff=2.0)
+def ssm_list_command_invocations(conn, **kwargs):
+    return conn.list_command_invocations(**kwargs)
+
+
+def main():
+    argument_spec = ec2_argument_spec()
```

Comment:
This is unnecessary, by default AnsibleAWSModule automatically merges the spec you pass it with ec2_argument_spec().